Tigger/Coordinates.py

```diff
@@ -389,9 +434,38 @@ def __init__(self, header):
             has_projection = True
             _Projector.__init__(self, ra0 * DEG, dec0 * DEG, has_projection=has_projection)
 
+        def check_angles(self, dec):
+            """Checks that angle is -90 > angle < 90. Adapted from astropy intrenals."""
+            try:
+                _check = dec * u.rad.to(u.deg)
+                if isinstance(_check, ndarray):
+                    with numpy.errstate(invalid='ignore'):
+                        invalid_angles = (numpy.any(_check < -90)
+                                          or numpy.any(_check > 90))
+                    return not invalid_angles
+                elif -90 <= _check <= 90:
+                    return True
+                else:
+                    return False
+            except:
+                return False
+
         def lm(self, ra, dec):
-            coord = SkyCoord(ra=ra * u.rad, dec=dec * u.rad)
-            coord_pixels = utils.skycoord_to_pixel(coords=coord, wcs=self.wcs, origin=0, mode='all')
+            if self.radesys == 'galactic':
+                coord = SkyCoord(l=ra * u.deg, b=dec * u.deg, frame=self.radesys)
+                coord = coord.transform_to('icrs')
+                coord_pixels = utils.skycoord_to_pixel(coords=coord, wcs=self.wcs, origin=0, mode='all')
+            elif self.radesys == 'geocentricmeanecliptic':
+                coord = SkyCoord(lon=ra * u.arcsec, lat=dec * u.arcsec, frame=self.radesys)
+                coord = coord.transform_to('icrs')
+                coord_pixels = utils.skycoord_to_pixel(coords=coord, wcs=self.wcs, origin=0, mode='all')
+                coord_pixels = np.array([coord_pixels[0], coord_pixels[1]])
+            elif self.check_angles(dec):
+                coord = SkyCoord(ra * u.rad, dec * u.rad, frame=self.radesys)
+                coord_pixels = utils.skycoord_to_pixel(coords=coord, wcs=self.wcs, origin=0, mode='all')
+            else:
+                raise RuntimeError(f"Angle not -90 deg > angle < 90 deg: {dec * u.rad.to(u.deg)}. Check Units, they may be incorrect.")
```

Comment:
This looks dangerous to me -- I'm not sure why you are flipping between radians, degrees and arcseconds. As far as I can tell above the CDELT is converted to radians for internal use.

According to FITS standard v 4.0 the units must conform to IAU style guide, and the accepted unit for angles are degrees:
![image](https://user-images.githubusercontent.com/7934586/169783332-48959a45-348d-40fb-ae50-76801aa03b42.png)

Reply:
This may be left over from when I was checking calculations in Tigger. SkyCoord internally is holding values as such:
```
galatic coord <SkyCoord (Galactic): (l, b) in deg
    (5.40178676, 7.2722058e-06)>
galatic coord to icrs <SkyCoord (ICRS): (ra, dec) in deg
    (269.48975689, -24.29066704)>
```
However let me look into this.

Reply:
@bennahugo - Something has gone awry since this comment and the verification https://github.com/ratt-ru/tigger/issues/127#issuecomment-1132873602

I am re-verifing things, please bear with me.

Reply:
@bennahugo - All fixed, something got mixed up, my bad. However through this process, I have reverified calculations against DS9 -  LTGM \o/